src/charm.py

```diff
@@ -428,7 +447,7 @@ def _common_exit_hook(self, update_ca_certs: bool = False) -> None:
         try:
             self.alertmanager_workload.update_config(self._render_manifest())
         except (ConfigUpdateFailure, ConfigError) as e:
-            self.unit.status = BlockedStatus(str(e))
+            self.last_error = str(e)
```

Comment:
This is interesting. I can see why you're storing state in this attribute here (and below) in the common exit hook. However, normally you'd query in collect-status the current state of things, rather than look at previously-set state attributes. Might be tricky for this though?